### PR TITLE
Manage submissions rebase to master

### PR DESCRIPTION
**Pull request by michellexliu**
### Summary generated by Reviewpad on 29 Jan 24 00:08 UTC
This pull request includes the following changes:

1. In the file `mkdocs.yml`, the `Admin Features` section is added under the `Nav` section.

2. In the file `README.md`, the `mkdir` command is modified. The `attachments/` directory is removed, while the `tmp/` directory remains unchanged.

3. In the file `spec/sets/submissions.rb`:
   - Line 9: The method `create_submissions_for_student` is called with incorrect arguments.
   - Line 22: An additional argument `version_number` is added to the method `AssessmentUserDatum.create_modulo_callbacks`.
   - Line 29: The argument `filename` is removed from the method `FactoryBot.create(:submission)`.
   - Line 34: The argument `problem` is removed from the method `FactoryBot.create(:score)`.

4. In the file `edit.html.erb`:
   - Several lines have replaced `f: f` with just `f:`, which seems to be an error.
   - Lines 3-21 handling error messages for `@course` and `@assessment` have been removed.
   - The `<div>` elements with ids `tab_basic`, `tab_handin`, `tab_penalties`, `tab_problems`, and `tab_advanced` have their corresponding `render` calls changed to `f:`.
   - The `<%= render "edit_basic", f: f %>`, `<%= render "edit_handin", f: f %>`, `<%= render "edit_penalties", f: f %>`, `<%= render "edit_problems", f: f %>`, and `<%= render "edit_advanced", f: f %>` calls have been replaced with just `f:`.

5. In the file `device_flow_requests_spec.rb`:
   - Line 51 and Line 57: The `device_code` parameter is missing in the `get :authorize` requests.

6. A migration file named "20231208083728_add_version_number_to_assessment_user_data.rb" is added. It adds a new column called "version_number" of type integer to the "assessment_user_data" table.

7. In the file `assessment/handout.rb`, changes are made to improve the validation and handling of the handout path in the `handout` method.

8. The migration file "20120204025713_add_tweak_type_to_submissions.rb" has changes to the `self.up` method.

9. A new file "moss.js" is added, which contains a function called `filterCourses` that filters a list of courses based on a given name.

10. The `test.rb` file in the `config/environments` directory has changes to configure the test environment.

11. Changes in the `watchlist_instance.rb` file include modifying and removing some arguments and modifying the `violation_info` argument in certain methods.

12. Changes made in the `annotations.scss` file include disabling Bootstrap styling, adding a CSS rule, and no other significant changes.

13. Changes in the `rails_helper.rb` file include modifying the temporary fix for chromedriver version issue, updating the driver setup for `headless_chrome`, and removing variables and updating the initialization of `Capybara::Selenium::Driver`.

14. The `attachments/_form.html.erb` file has undergone changes, including JavaScript code, conditional blocks, and updates to form elements.

15. The `_form.html.erb` file in the `app/views/autograders` directory has changes to JavaScript, form elements, and the delete link.

16. The `bin/rake` file has modifications to load the `spring` file and update the `require` statement for the `rake` library.

17. Changes in the `app/views/announcements/_announcementFields.html.erb` file include updated indentation, added `display_name` attributes, removal of a TODO comment, and replacing the submit button text.

Please review these changes and provide feedback.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have run rubocop and erblint for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting
- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://docs.autolabproject.com/)
- [ ] I have updated the documentation accordingly, included in this PR

## Other issues / help required

If unsure, feel free to submit first and we'll help you along.

### CODE REVIEW

File: app/views/submissions/index.html.erb [hunk 1]
```diff
@@ -19,43 +19,62 @@
...
+  <div class="buttons-spacing">
     <%= link_to "<i class='material-icons left submissions-icons'>file_download</i>Download Final Submissions".html_safe,
-        downloadAll_course_assessment_submissions_path(@course, @assessment, final: true),
-        {:title=>"Download final submissions from each student",
-         :class=>"btn submissions-main"} %>
-    </div>
-
-    <div class="buttons-spacing"> <!-- TODO: Change functionality to filter for missing submissions -->
+                downloadAll_course_assessment_submissions_path(@course, @assessment),
+                { title: "Down all submissions from each student",
```
Comment (damianhxy) [inline app/views/submissions/index.html.erb:31]: Nit: typo "Download"

File: app/views/submissions/index.html.erb [hunk 2]
```diff
@@ -19,43 +19,62 @@
...
-    <div class="buttons-spacing">
+                downloadAll_course_assessment_submissions_path(@course, @assessment, final: true),
+                { title: "Download the most recent submission from each student",
+                  class: "btn submissions-main" } %>
+  </div>
+  <div class="buttons-spacing">
     <%= link_to "<i class='material-icons left submissions-icons'>event</i>Manage Extensions".html_safe,
-        [@course, @assessment, :extensions],
-        {:class=>"btn submissions-main"} %>
+                missing_course_assessment_submissions_path(@course, @assessment),
```
Comment (damianhxy) [inline app/views/submissions/index.html.erb:42]: These three buttons are linked to the wrong path
- download final -> download all
- download all -> (missing)
- missing submissions -> download final
- manage extensions -> missing submissions
- (missing) -> manage extensions

**Review by damianhxy — CHANGES_REQUESTED** [review]
Followed testing steps from #1893, #1886, and #1990

See comment re: issue with the buttons at the top

File: app/views/submissions/index.html.erb [hunk 3]
```diff
@@ -19,43 +19,62 @@
...
-    <div class="buttons-spacing">
-    <%= link_to "<i class='material-icons left submissions-icons'>add</i>Create Submission".html_safe,
-        new_course_assessment_submission_path(@course, @assessment),
-        {:title=>"Create a new submission for a student, with an option to submit a handin file on their behalf",
-         :class=>"btn submissions-main"} %>
-    </div>
-
-    <div class="buttons-spacing">
+  <div class="buttons-spacing">
+    <%= link_to "<i class='material-icons left submissions-icons'>add</i>Create Submission".html_safe.html_safe,
```
Comment (damianhxy) [inline app/views/submissions/index.html.erb:23]: Nit: `.html_safe` used twice

File: app/assets/javascripts/manage_submissions.js [hunk 4]
```diff
@@ -236,4 +236,4 @@ $(document).ready(function () {
     e.stopPropagation();
   });
 
-});
+})
```
Comment (damianhxy) [inline app/assets/javascripts/manage_submissions.js:239]: Nit: undo this removal

**Review by damianhxy — CHANGES_REQUESTED** [review]
Nits and some regressions

File: app/views/submissions/index.html.erb [hunk 5]
```diff
@@ -20,26 +20,26 @@
...
+    <%= link_to "<i class='material-icons left submissions-icons'>add</i>Create Submission".html_safe,
                 new_course_assessment_submission_path(@course, @assessment),
                 { title: "Create a new submission for a student, with an option to submit a handin file on their behalf",
                   class: "btn submissions-main" } %>
   </div>
   <div class="buttons-spacing">
     <%= link_to "<i class='material-icons left submissions-icons'>file_download</i>Download Final Submissions".html_safe,
                 downloadAll_course_assessment_submissions_path(@course, @assessment),
-                { title: "Down all submissions from each student",
+                { title: "Download final submissions from each student",
```
[the remark this answers is not in the document]
Comment (michellexliu) [reply app/views/submissions/index.html.erb:31]: https://www.figma.com/proto/XbxY3zscou3HAcc9maC8sl/%5BAutolab%5D-Final-Manage-Submissions-UI?node-id=330-862&scaling=contain&page-id=0%3A1&starting-point-node-id=319%3A1619&hotspot-hints=0

i think in lynn's design we're getting rid of the download all button; it seems that there's not much use case for downloading all submissions for every student
image

in the new design you can download all of the submissions for an individual student by clicking on the checkmark and hitting "download selected" (feature to be implemented in a future PR once grouping is implemented)
image

[the remark this answers is not in the document]
Comment (damianhxy) [reply app/views/submissions/index.html.erb:31]: Ahh okay

**Review by damianhxy — APPROVED** [review]
LGTM

**Pull request merged**